Source/Scene/MetadataTable.js

```diff
@@ -0,0 +1,291 @@
+  Check.typeOf.string("propertyId", propertyId);
+  //>>includeEnd('debug');
+
+  var property = this._properties[propertyId];
+
+  if (!defined(property)) {
+    return;
+  }
+
+  //>>includeStart('debug', pragmas.debug);
+  var classProperty = this._class.properties[propertyId];
+  var errorMessage = classProperty.validate(value);
+  if (defined(errorMessage)) {
+    throw new DeveloperError(errorMessage);
+  }
+  //>>includeEnd('debug');
+
+  value = MetadataEntity.unnormalize(this._class, propertyId, value);
+
+  property.set(index, value);
+};
+
+/**
+ * Returns a copy of the value of the property with the given semantic.
+ *
+ * @param {Number} index The index of the entity.
+ * @param {String} semantic The case-sensitive semantic of the property.
+ * @returns {*} The value of the property or <code>undefined</code> if the property does not exist.
+ *
+ * @exception {DeveloperError} index is required and between zero and count - 1
+ */
+MetadataTable.prototype.getPropertyBySemantic = function (index, semantic) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.string("semantic", semantic);
+  //>>includeEnd('debug');
+
+  if (defined(this._class)) {
+    var property = this._class.propertiesBySemantic[semantic];
+    if (defined(property)) {
+      return this.getProperty(index, property.id);
+    }
+  }
+  return undefined;
+};
+
+/**
+ * Sets the value of the property with the given semantic.
+ *
+ * @param {Number} index The index of the entity.
+ * @param {String} semantic The case-sensitive semantic of the property.
+ * @param {*} value The value of the property that will be copied.
+ *
+ * @exception {DeveloperError} index is required and between zero and count - 1
+ * @exception {DeveloperError} value does not match type
+ * @exception {DeveloperError} value is out of range for type
+ * @exception {DeveloperError} Array length does not match componentCount
+ */
+MetadataTable.prototype.setPropertyBySemantic = function (
+  index,
+  semantic,
+  value
+) {
+  //>>includeStart('debug', pragmas.debug);
+  Check.typeOf.string("semantic", semantic);
+  //>>includeEnd('debug');
+
+  if (defined(this._class)) {
+    var property = this._class.propertiesBySemantic[semantic];
+    if (defined(property)) {
+      this.setProperty(index, property.id, value);
+    }
+  }
+};
+
+function getDefault(classDefinition, propertyId) {
+  if (defined(classDefinition)) {
+    var classProperty = classDefinition.properties[propertyId];
+    if (defined(classProperty) && defined(classProperty.default)) {
+      if (classProperty.type === MetadataType.ARRAY) {
+        return classProperty.default.slice(); // clone
+      }
+      return classProperty.default;
+    }
+  }
+}
+
+function checkIndex(table, index) {
+  var count = table._count;
+  if (!defined(index) || index < 0 || index >= count) {
+    var maximumIndex = count - 1;
+    throw new DeveloperError(
+      "index is required and between zero and count - 1 (" + maximumIndex + ")."
```

Comment:
Maybe something like this?

```suggestion
      "index is required and between zero and count - 1. Actual value: " + maximumIndex;
```

I could see someone getting confused if it formatted it like `between zero and count - 1 (2)` (as juxtaposition can mean multiplication)